components/interactive_dialog/dialog_element/dialog_element.tsx

```diff
@@ -15,33 +14,47 @@ import AutocompleteSelector from 'components/autocomplete_selector';
 import ModalSuggestionList from 'components/suggestion/modal_suggestion_list.jsx';
 import BoolSetting from 'components/widgets/settings/bool_setting';
 import RadioSetting from 'components/widgets/settings/radio_setting';
+import Provider from 'components/suggestion/provider';
 
 const TEXT_DEFAULT_MAX_LENGTH = 150;
 const TEXTAREA_DEFAULT_MAX_LENGTH = 3000;
 
-export default class DialogElement extends React.PureComponent {
-    static propTypes = {
-        displayName: PropTypes.string.isRequired,
-        name: PropTypes.string.isRequired,
-        type: PropTypes.string.isRequired,
-        subtype: PropTypes.string,
-        placeholder: PropTypes.string,
-        helpText: PropTypes.string,
-        errorText: PropTypes.node,
-        maxLength: PropTypes.number,
-        dataSource: PropTypes.string,
-        optional: PropTypes.bool,
-        options: PropTypes.arrayOf(PropTypes.object),
-        value: PropTypes.any,
-        onChange: PropTypes.func,
-        autoFocus: PropTypes.bool,
-        actions: PropTypes.shape({
-            autocompleteChannels: PropTypes.func.isRequired,
-            autocompleteUsers: PropTypes.func.isRequired,
-        }).isRequired,
+type Props = {
+    displayName: string;
+    name: string;
+    type: any;
+    subtype?: string;
+    placeholder?: string;
+    helpText?: string;
+    errorText?: React.ReactNode;
+    maxLength?: number;
+    dataSource?: string;
+    optional?: boolean;
+    options?: {text: string;value: string}[];
+    value?: string;
+    onChange?: (name: string, id: string) => void;
+    autoFocus?: boolean;
+    actions: {
+        autocompleteChannels: () => void;
+        autocompleteUsers: () => void;
```

Comment:
is this the right signature for this functions?

Reply:
That'd probably be a good idea since it will make it easier to convert the providers to TypeScript eventually as well.